src/main/scala/ai/privado/tagger/S3Tagger.scala

```diff
@@ -0,0 +1,130 @@
+package ai.privado.tagger
+
+import ai.privado.cache.{DatabaseDetailsCache, S3DatabaseDetailsCache}
+import ai.privado.languageEngine.java.language.NodeStarters
+import ai.privado.model.DatabaseDetails
+import io.shiftleft.codepropertygraph.generated.{Cpg, Languages, Operators}
+import org.slf4j.LoggerFactory
+import io.shiftleft.codepropertygraph.generated.nodes.{Call, Method}
+import io.shiftleft.semanticcpg.language.*
+import io.shiftleft.codepropertygraph.generated.nodes.JavaProperty
+import org.slf4j.LoggerFactory
+import overflowdb.traversal.*
+
+import scala.collection.mutable.ListBuffer
+
+class S3Tagger(cpg: Cpg) extends PrivadoSimpleCpgPass(cpg) {
+
+  private val logger          = LoggerFactory.getLogger(getClass)
+  private val bucketPattern   = "(?!(^xn--|.+-s3alias$))^[a-z0-9][a-z0-9-]{1,61}[a-z0-9]$"
+  private val readRuleId      = "Storages.AmazonS3.Read"
+  private val writeRuleId     = "Storages.AmazonS3.Write"
+  private val readWriteRuleId = "Storages.AmazonS3.ReadAndWrite"
+
+  override def run(builder: DiffGraphBuilder): Unit = {
+    val readBucketNames      = ListBuffer[String]()
+    val writeBucketNames     = ListBuffer[String]()
+    val readWriteBucketNames = ListBuffer[String]()
+
+    if (cpg.metaData.language.head == Languages.JAVASRC) {
```

Comment:
I see we are using conditions, can we not have an abstract S3BaseTagger, whose `generateParts` functions can be defined at the language level.

Effectively having `JavaS3Tagger` and `PythonS3Tagger` extending `S3BaseTagger`

Reply:
Ok, yes, this is a way. It will also avoid needing to run the tagger for each language as I am doing now

Reply:
Since this will eventually be overhauled and added to a totally new system, can we let this in? We see S3 tagging used in its current state for half a month more before we restructure this and move this in a more comprehensive asset tagging

Reply:
Better to do it now, as things move slowly once in backlog